src/arrayMethodReduce.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator;
+    let initialIndex = 0;
+
+    if (arguments.length === 1) {
+      accumulator = this[0];
+      initialIndex = 1;
+    } else if (arguments.length === 2 && !initialValue && initialValue !== 0) {
+      accumulator = String(initialValue);
+    } else {
+      accumulator = initialValue;
+    }
```

Comment:
Try to use only one construction if instead if, else if, else.
You can check if the length is less than 2